interval/lift/lift.go

```diff
@@ -132,3 +134,66 @@ func StrictBorderCheck(c chain.Chain, i interval.Interval) bool {
 	_, border = chain.TPosToQPos(c, i.GetChromEnd()-1) //interval ranges are open right so we want ChromEnd - 1
 	return border
 }
+
+//everything below is for lifting with AXT
+
+// refCoordToRefIdx is a helper function for LiftCoordinatesWithAxt which identifies the start and end indices in the Axt reference multifa record that corresponds to the interval to be lifted
+func refCoordToRefIdx(a axt.Axt, region interval.Interval) (start, end int) {
+	var stopLoop int
+	for i := range a.RSeq {
+		if stopLoop >= (region.GetChromStart() - (a.RStart - 1)) {
+			break
+		}
+		if a.RSeq[i] != 10 {
+			stopLoop++
+		}
+		start++
+	}
+	end = start
+	stopLoop = 0
+	for i := start; i < len(a.RSeq); i++ {
+		if stopLoop >= (region.GetChromEnd() - region.GetChromStart()) {
+			break
+		}
+		if a.RSeq[i] != 10 {
+			stopLoop++
+		}
+		end++
+	}
+	return start, end
+}
+
+// translateCoord is a helper function for LiftCoordinatesWithAxt which takes the indices of the multifa for the region to be lifted and put them in query coordinates
+func translateCoord(a axt.Axt, start, end int) (newStart, newEnd int) {
+	newStart = (a.QStart - 1) + dna.CountBasesNoGaps(a.QSeq[0:start])
+	newEnd = newStart + dna.CountBasesNoGaps(a.QSeq[start:end])
+	return newStart, newEnd
+}
+
+// LiftCoordinatesWithAxt takes an Axt record, a lift-compatible interval and chromosome size for the query (in case of a minus strand alignment), and lifts the interval from the reference
+// coordinates to the query coordinates. The interval must be completely contained within the reference Axt region. The output is chromosome name, start (0-based) and end coordinates
+// of the lifted interval
+func LiftCoordinatesWithAxt(a axt.Axt, region Lift, Qsize int) (chrom string, start, end int) {
+	if !checkCompatability(a, region) {
+		log.Fatalf("The interval you are trying to lift is not entirely within the axt refernce coordinates.")
+	}
+	chrom = a.QName
+	refStart, refEnd := refCoordToRefIdx(a, region)
+	newStart, newEnd := translateCoord(a, refStart, refEnd)
+	start, end = newStart, newEnd
+	if !a.QStrandPos {
```

Comment:
Note to self: if query strand is -, lastz generates the entire reverse complemented contig, then does alignment and gets query ref pos, so in order to get the correct strand (non reverse complemented) contig pos (coordinates), need to know query chrom size (contig size)